torch/testing/_internal/common_utils.py

```diff
@@ -901,14 +902,21 @@ def check_disabled(test_name):
 
         def read_and_process():
             url = 'https://raw.githubusercontent.com/pytorch/test-infra/master/stats/disabled-tests.json'
-            contents = urlopen(url, timeout=1).read().decode('utf-8')
             the_response = fetch_and_cache(".pytorch-disabled-tests", url)
             for item in the_response['items']:
                 title = item['title']
                 key = 'DISABLED '
                 if title.startswith(key):
                     test_name = title[len(key):].strip()
-                    _disabled_test_from_issues[test_name] = item['html_url']
+                    body = item['body']
+                    platforms_to_skip = []
+                    key = 'platforms:'
+                    for line in body.splitlines():
+                        line = line.lower()
+                        if line.startswith(key):
+                            pattern = re.compile(r"^\s+|\s*,\s*|\s+$")
```

Comment:
question from a regex noob: do the `^` and `$` matter here? do they apply only to the first and third alternatives in this regex, or to the entire thing?

Reply:
I think you got confused with the negate usage of `^`. Here, using `^` and `$` at the start/end of a regex meaning it should match the entire string https://www3.ntu.edu.sg/home/ehchua/programming/howto/Regexe.html#zz-1.4

Reply:
The entire thing, but yea I think you're right that these wouldn't matter since we splitlines anyway